spyder/utils/encoding.py

```diff
@@ -301,7 +310,12 @@ def read(filename, encoding='utf-8'):
     Read text from file ('filename')
     Return text and encoding
     """
-    text, encoding = decode( open(filename, 'rb').read() )
+    if filename.endswith(('.py', '.pyw', '.ipy')):
```

Comment:
```suggestion
    if filename.endswith(('.py', '.pyw', '.ipy', '.pyi')):
```

That's the extension for typing files.